src/components/Product.js

```diff
@@ -2,37 +2,43 @@ import React from "react";
 import Rating from "./Rating";
 import "../style/product.css";
 import {useCart} from '../context/cart-context'
+import {useWishList} from '../context/wishlist-context'
 
-function Product({img, name, type, rating, price, data}) {
+function Product({data}) {
 
   const { dispatch } = useCart();
+  const { wishListDispatch } = useWishList();
+
 
   return (
     <div className="product_item">
-      <img className="img_responsive" src={img} alt=""></img>
+      <img className="img_responsive" src={data.img} alt=""></img>
       <div className="item_desc">
-        <h3 className="item_desc_heading">{name}</h3>
-        <p>{type}</p>
-        <p>Rs: {price}</p>
+        <h3 className="item_desc_heading">{data.name}</h3>
+        <p>{data.product}</p>
+        <p>Rs: {data.price}</p>
       </div>
 
       <div className="rating">
-        <Rating rating={rating} />
+        <Rating rating={data.rating} />
       </div>
 
       <div>
-        <button className="btn wishlist_btn primary_solid_btn" type="submit" 
+        <button className="btn cart_btn primary_solid_btn" type="submit" 
         onClick={() => {
             dispatch({
               type: "ADD_TO_CART",
-              totalPrice: price,
+              totalPrice: data.price,
               payload: data,
```

Comment:
You can keep the `data` as it is in prop and destructure it inside the component. This way you will have the benefit of destructing as well as you can use the `data` object as usual